### PR TITLE
Add an npm module project for the typescript recipe runtime classes

### PR DESCRIPTION
**Pull request by kormide**
@jbedard

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -3,3 +3,4 @@
 */target/*
 *.swp
 *.iml
+*/node_modules
```
Comment (jbedard) [inline .gitignore]: slash at the end like `target`?

File: recipe-ts-runtime/package.json [hunk 2]
```diff
@@ -0,0 +1,18 @@
+{
+  "name": "recipe-ts-runtime",
+  "version": "0.0.1",
+  "description": "Typescript run-time library for the Recipe framework",
```
Comment (jbedard) [inline recipe-ts-runtime/package.json]: TypeScript has a capital S :P

File: recipe-ts-runtime/pom.xml [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+                <executions>
+                    <execution>
+                        <id>install-node-and-yarn</id>
+                        <phase>initialize</phase>
+                        <goals>
+                            <goal>install-node-and-yarn</goal>
+                        </goals>
+                        <configuration>
+                            <nodeVersion>v6.10.0</nodeVersion>
```
Comment (jbedard) [inline recipe-ts-runtime/pom.xml]: These versions seem ancient!

File: recipe-ts-runtime/tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+        "target": "es6",
+        "module": "commonjs",
+
+        "lib": ["es6"],
+
+        "alwaysStrict": true,
+        "noUnusedLocals": true,
+        "noUnusedParameters": false,
+        "removeComments": false,
+        "sourceMap": false,
```
Comment (jbedard) [inline recipe-ts-runtime/tsconfig.json]: probably want `true` so it can be debugged

File: recipe-ts-runtime/src/Recipe.ts [hunk 5]
```diff
@@ -0,0 +1,4 @@
+import {Ingredient} from "./Ingredient";
```
Comment (jbedard) [inline recipe-ts-runtime/src/Recipe.ts]: All the other imports have spaces around the `{`

File: recipe-ts-runtime/package.json [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/kormide/recipe"
+  },
+  "license": "MIT",
+  "private": false,
+  "dependencies": {
+    "tslint": "4.5",
+    "typescript": "2.2"
```
Comment (jbedard) [inline recipe-ts-runtime/package.json]: this is a pretty old version, use 2.6

**Comment by kormide** [comment]
Updated

**Comment by kormide** [comment]
Updated

**Pull request merged**